### PR TITLE
Remove Python 2 style int division

### LINKED ISSUE #202: Remove Python 2 style int division
The config setting `int_division` allows for Python 2 style integer division. Python 2 was EOL and we should stop supporting this.

The `int_division` config setting is accessed in 6 places of `basic.py`.

### PR DESCRIPTION
**Pull request by LegrandNico**
This PR drops support for Python 2 style int division in `scalar` and `tensor` (#202). It removes functions choosing division type from data type. The division (`int_div` or `true_div`) type should now be explicitly provided or will use the default Python 3 type.

* remove `int_division` parameter from `configdefault.py`
* remove `int_or_true_div()` and `div_proxy` from `theano.scalar.basic.py`, only use `int_div` or `true_div` directly.

### CODE REVIEW

**Review by twiecki — DISMISSED** [review]

File: theano/scalar/basic.py [hunk 1]
```diff
@@ -798,10 +797,10 @@ def __mul__(self, other):
         return mul(self, other)
 
     def __truediv__(self, other):
-        return div_proxy(self, other)
+        return true_div(self, other)
 
-    def __div__(self, other):
-        return div_proxy(self, other)
+    def __intdiv__(self, other):
+        return int_div(self, other)
```
Comment (michaelosthege) [inline theano/scalar/basic.py:803]: Can one of the old tests be used to cover this?

**Review by michaelosthege — CHANGES_REQUESTED** [review]
Awesome, so many deleted lines!

I saw two details that maybe should be adressed.

File: tests/tensor/test_basic.py [hunk 2]
```diff
@@ -5495,24 +5482,14 @@ def numpy_i_scalar(dtype):
...
-                                    )
-                                    # This is the expected behavior.
-                                    continue
+
+                                theano_dtype = op(
+                                    theano_args[0](a_type), theano_args[1](b_type)
+                                ).type.dtype
+                                # Should have crashed if it is an integer
+                                # division and `config.int_division` does
+                                # not allow it.
```
[the remark this answers is not in the document]
Comment (brandonwillard) [reply tests/tensor/test_basic.py:5491]: Yes, the comment needs to be removed.

File: theano/scalar/basic.py (hunk 1 above)
Comment (brandonwillard) [reply theano/scalar/basic.py:803]: What exactly is `__intdiv__`?  Shouldn't this actually be `__div__`?  If not, where's the `__div__` implementation?

**Review by brandonwillard — CHANGES_REQUESTED** [review]

**Review by brandonwillard — CHANGES_REQUESTED** [review]
Looks good, but it needs to be rebased (and have some conflicts resolved).

**Review by michaelosthege — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks everyone!

**Comment by brandonwillard** [comment]
This PR doesn't appear to have been rebased; it added a merge commit instead.  As a general rule, we don't want merge commits in feature/PR branches (especially not from the main branch).

**Comment by LegrandNico** [comment]
Sorry, I thought this was also working. Do you want me to do a proper rebase before we merge?

**Comment by brandonwillard** [comment]
No problem, it's already merged and nothing is broken.  We just don't want a complicated and difficult-to-use (e.g. revert, bisect, etc.) Git history, so we should always rebase onto `origin/master`&mdash;instead of adding merge commits&mdash;before merging into `master`.

**Pull request merged**